tests/ci/integration/libgit2_patch/0001-Support-for-AWS-LC.patch

```diff
@@ -0,0 +1,87 @@
+From 399b74c9a4fd6caa55118f6f9ffdb93808aa0fbc Mon Sep 17 00:00:00 2001
+From: Justin Smith <justsmth@amazon.com>
+Date: Tue, 25 Feb 2025 11:50:34 -0500
+Subject: [PATCH] Support for AWS-LC
+
+---
+ deps/ntlmclient/crypt_openssl.c |  8 +++++++-
+ deps/ntlmclient/crypt_openssl.h | 29 +++++++++++++++++++++++++++++
+ 2 files changed, 36 insertions(+), 1 deletion(-)
+
+diff --git a/deps/ntlmclient/crypt_openssl.c b/deps/ntlmclient/crypt_openssl.c
+index 3bec27259..abdb46322 100644
+--- a/deps/ntlmclient/crypt_openssl.c
++++ b/deps/ntlmclient/crypt_openssl.c
+@@ -44,7 +44,7 @@ NTLM_INLINE(void) HMAC_CTX_free(HMAC_CTX *ctx)
+ 
+ #endif
+ 
+-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)) || \
++#if (OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_IS_AWSLC)) || \
+ 	(defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER >= 0x03050000fL) || \
+ 	defined(CRYPT_OPENSSL_DYNAMIC)
+ 
+@@ -214,8 +214,14 @@ bool ntlm_hmac_md5_init(
+ 
+ 	ntlm->crypt_ctx.hmac_ctx_cleanup_fn(ntlm->crypt_ctx.hmac);
+ 
++#if defined(OPENSSL_IS_AWSLC)
++	ntlm->crypt_ctx.hmac_ctx_reset_fn(ntlm->crypt_ctx.hmac);
++	return ntlm->crypt_ctx.hmac_init_ex_fn(ntlm->crypt_ctx.hmac, key, key_len, md5, NULL);
++#else
+ 	return ntlm->crypt_ctx.hmac_ctx_reset_fn(ntlm->crypt_ctx.hmac) &&
+ 	       ntlm->crypt_ctx.hmac_init_ex_fn(ntlm->crypt_ctx.hmac, key, key_len, md5, NULL);
++#endif
++
+ }
+ 
+ bool ntlm_hmac_md5_update(
+diff --git a/deps/ntlmclient/crypt_openssl.h b/deps/ntlmclient/crypt_openssl.h
+index 8654027db..c18df6848 100644
+--- a/deps/ntlmclient/crypt_openssl.h
++++ b/deps/ntlmclient/crypt_openssl.h
+@@ -60,6 +60,34 @@ typedef struct hmac_ctx_st {
+ } HMAC_CTX;
+ #endif
+ 
++#if defined(OPENSSL_IS_AWSLC)
```

Comment:
nit: suggest adding a comment explaining how this differs from the other `ntlm_crypto_ctx` definition and why AWS-LC needs an alternate